src/server/metadata/cmds/cmds.go

```diff
@@ -18,15 +21,21 @@ func Cmds(mainCtx context.Context, pachctlCfg *pachctl.Config) []*cobra.Command
 	var commands []*cobra.Command
 
 	editMetadata := &cobra.Command{
+		Use:   `{{alias}} [<object type: project> <object picker> <operation: add|edit|delete|set> <data: key=value, key, '{"key":"value","key2":"value2"}'>]...`,
 		Short: "Edits an object's metadata",
 		Long:  "Edits an object's metadata",
-		Run: cmdutil.RunFixedArgs(0, func(args []string) error {
+		Run: cmdutil.Run(func(args []string) error {
+			req, err := parseEditMetadataCmdline(args)
+			if err != nil {
+				return errors.Wrap(err, "parse cmdline")
+			}
+
 			c, err := pachctlCfg.NewOnUserMachine(mainCtx, false)
```

Comment:
This uses the `mainCtx` from the call to `Cmds`; I think that you want to use `cmd.Context()` from the running Cobra command (with `cmdutil.RunCmd`).

Reply:
Done, thanks to your refactor 🎉